stdlib/public/core/StringInterpolation.swift

```diff
@@ -10,11 +10,84 @@
 //
 //===----------------------------------------------------------------------===//
 
-extension String : _ExpressibleByStringInterpolation {
-  /// Creates a new string by concatenating the given interpolations.
-  ///
+/// Represents a string literal with interpolations while it is being built up.
+/// 
+/// Do not create an instance of this type directly. It is used by the compiler
+/// when you create a string using string interpolation. Instead, use string
+/// interpolation to create a new string by including values, literals,
+/// variables, or expressions enclosed in parentheses, prefixed by a
+/// backslash (`\(`...`)`).
+///
+///     let price = 2
+///     let number = 3
+///     let message = "If one cookie costs \(price) dollars, " +
+///                   "\(number) cookies cost \(price * number) dollars."
```

Comment:
This example was written more than two years ago (before [SE-0168][] and [SE-0182][]).

Would it be any better with a multi-line string?

```swift
///     let message = """
///                   If one cookie costs \(price) dollars, \
///                   \(number) cookies cost \(price * number) dollars.
///                   """
```

Unless there is a compile-time optimization for the concatenation?

[SE-0168]: <https://github.com/apple/swift-evolution/blob/master/proposals/0168-multi-line-string-literals.md>
[SE-0182]: <https://github.com/apple/swift-evolution/blob/master/proposals/0182-newline-escape-in-strings.md>

Reply:
I'll change these throughout.